src/errorpage.h

```diff
@@ -287,10 +322,42 @@ class TemplateFile
      */
     bool tryLoadTemplate(const char *lang);
 
+    SBuf textBuf; ///< A Buffer to store the template
     bool wasLoaded; ///< True if the template data read from disk without any problem
     String errLanguage; ///< The error language of the template.
     String templateName; ///< The name of the template
     err_type templateCode; ///< The internal code for this template.
+    SBuf lastTemplateFile; ///< The last used path
+};
+
+class ErrTextValidator {
+public:
+    ErrTextValidator() {}
+    ErrTextValidator(const char *aName) : name_(aName) {}
+
+    ErrTextValidator &useCfgContext(const char *filename, int lineNo, const char *line);
+
+    ErrTextValidator &useFileContext(const char *templateFilename);
+
+    ErrTextValidator &warn(int level) { warn_ = level; return *this; }
+
+    ErrTextValidator &fatal(bool abort = true) { fatal_ = abort; return *this; }
```

Comment:
I would drop the formal parameter because `fatal()` and `throws()` are mutually exclusive. You cannot meaningfully turn then on/off individually. The parameter will also become impossible to support if you switch to the return/quit/throw enum for storing the action as discussed in another change request.